deploy-agent/deployd/common/single_instance.py

```diff
@@ -32,7 +32,13 @@ def __init__(self):
         appname = 'deploy-agent'
         lockfile_name = '.{}.lock'.format(appname)
         self._create_lock_dir()
-        lockfile_path = os.path.join(LOCKFILE_DIR, lockfile_name)
+        # Backward compatibility as old deploy agent versions use lock file in /tmp. 
+        # Use the old lock file if it exists 
+        tmp_lockfile_path = os.path.join(tempfile.gettempdir(), lockfile_name)
+        if os.path.exists(tmp_lockfile_path):
+            lockfile_path = tmp_lockfile_path
+        else: 
```

Comment:
If there is a lock file under `/tmp`, does it mean it will be used indefinitely? 

What about delete the old lock file if it's not locked instead?

Reply:
the old one will be used until hosts are rotated with a newer ami containing a new deploy agent version. I thought about it but eventually we want hosts to be refreshed with latest amis.